### DISCUSSION TITLE
Trying to install a prouct feature with Oomph results in no branding

### DISCUSSION THREAD

**Discussion by A7exSchin** [eclipse-oomph/oomph#10] [General]
Hi there Oomph Team!

Firstly, I greatly appreciate the project. For my Bachelors Thesis I am currently developing a tool that manages Eclipse Product installations (done with the Eclipse Installer).

One of the requirements is that my tool is able to manage an Eclipse Application called [Palladio Simulator](https://www.palladio-simulator.com/home/). For this purpose I created a product catalog and product setup files.

These are contained inside the Github repository https://github.com/A7exSchin/bootPalladio

Now to the question:
I install only the product feature on the Eclipse Platform Release 2021-12. The prouct feature includes the Branding Feature. Nonetheless, it does not install any splash-art, icon or Naming (palladio.exe instead of eclipse.exe). Why is this? Is this even possible with the Eclipse Installer?

Thanks for any help!

P.S: Anyone interested in the progress of my tool can check it out [here](https://eim.a7schin.dev)

**Comment by merks** [comment]
There must also be a product installable unit.  E.g.,
```
<unit id='org.eclipse.sdk.ide' version='4.28.0.I20230605-0440'>
  <update id='org.eclipse.sdk.ide' range='[4.0.0,4.28.0.I20230605-0440)' severity='0' description='An update for 4.x generation Eclipse SDK.'/>
  <properties size='5'>
    <property name='org.eclipse.equinox.p2.name' value='Eclipse SDK'/>
    <property name='org.eclipse.equinox.p2.type.product' value='true'/>
    <property name='org.eclipse.equinox.p2.type.group' value='true'/>
    <property name='org.eclipse.equinox.p2.description' value='4.28 Release of the Eclipse SDK.'/>
    <property name='org.eclipse.equinox.p2.provider' value='Eclipse.org'/>
  </properties>
```
So look for that unit specifically with ` <property name='org.eclipse.equinox.p2.type.product' value='true'/>` and be sure that it is being installed.

**Reply by A7exSchin** [reply]
Thanks for the reply on Sundays! I am not sure where to put that part of XML code. I replaced the `<requirement>` part for the `org.palladiosimulator.product.feature.feature.group` with the `<unit>` part, but it does not seem to recognize it to be installed.

Given I understand everythign correctly, `org.palladiosimulator.product.feature.feature.group` is the product feature IU.  Currently i have it set up like this 

```
<setupTask
    xsi:type="setup.p2:P2Task"
    label="Palladio Specific">
    <requirement name="org.palladiosimulator.product.feature.feature.group"/>
    <unit id="org.palladiosimulator.product.feature.feature.group">
        <update
            id="org.palladiosimulator.product.feature.feature.group"
            range="[5.0.0,6.0.0)"
            severity="0"
            description="Product installation for Palladio" />
        <properties size="5">
            <property
	        name="org.eclipse.equinox.p2.name"
	        value="Palladio Bench" />
            <property
	        name="org.eclipse.equinox.p2.type.product"
	        value="true" />
            <property
	        name="org.eclipse.equinox.p2.type.group"
	        value="true" />
            <property
	        name="org.eclipse.equinox.p2.description"
	        value="5.1.0 Release of the Palladio Bench" />
            <property
	        name="org.eclipse.equinox.p2.provider"
	        value="palladiosimulator.org" />
        </properties>
    </unit>
</setupTask>
```

How do I specify the unit part correctly? Is it only possible with XML or is there an option in the Setup Editor? Do I need a requirement part and the unit code? Does it need to be contained within the setup tasks?

I am sorry, compared to most I here, I consider myself a beginner when it comes to advanced Setup Configuration.

Thanks again!

**Reply by A7exSchin** [reply]
After researching some more I figured, that the `<unit>` part should not be part of my installation Setup File, but the target definition of the Palladio Target? If so, I do not have access to that.

So I guess it is not possible to tell Oomph that this IU is a product feature and it should be installed as one seperately?

**Reply by merks** [reply]
You should typically pretty much always use the structured editor to edit *.setup resources.  These instructions should help you author the p2 director task without to much effort:

https://wiki.eclipse.org/Eclipse_Oomph_Authoring#How_to_install_Eclipse_plugins_using_the_P2_Director_and_Repository_Explorer

Here's an example where I show all the IUs (see underlined controls) not just the top-level features, and find the product IU (double click to see the details).  Drag and drop that product IU onto the p2 directory task.

**Reply by A7exSchin** [reply]
Thank you very much! The help led me to view the repository with the proper settings, where I found the Product IU and installing that together with the feature.feature.group made everything work flawlessly!
